bblean/bitbirch.py

```diff
@@ -951,10 +951,10 @@ def dump_assignments(
                 f"Len of the provided smiles {len(smiles)}"
                 f" must match the number of fitted fingerprints {self.num_fitted_fps}"
             )
-        if not smiles:
+        if smiles is None:
             df = pd.DataFrame({"assignments": assignments})
         if smiles.size:
-            df["smiles"] = smiles
+            df = pd.DataFrame({"smiles": smiles, "assignments": assignments})
```

Comment:
Now assign the smiles if they are passed
```suggestion
            df["smiles"] = smiles
```